migration/007_schema_fix.sql

```diff
@@ -0,0 +1,114 @@
+-- +migrate Up
+
+-- +migrate StatementBegin
+DROP TABLE peer_claim_checkpoint;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+DROP TABLE claim_checkpoint;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+DROP TABLE peer_claim;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+DROP TABLE peer;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+DROP TABLE input_address;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+DROP TABLE output_address;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+ALTER TABLE address
+  DROP COLUMN tag,
+  DROP COLUMN tag_url,
+  CHANGE COLUMN created created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  CHANGE COLUMN modified modified_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+ALTER TABLE output
+    DROP COLUMN hash160,
+  CHANGE COLUMN created created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  CHANGE COLUMN modified modified_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+ALTER TABLE transaction
+    DROP COLUMN value,
+  CHANGE COLUMN created created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  CHANGE COLUMN modified modified_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+ALTER TABLE transaction_address
+    DROP COLUMN latest_transaction_time;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+ALTER TABLE block
+    DROP COLUMN median_time,
+  DROP COLUMN target,
+CHANGE COLUMN created created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+CHANGE COLUMN modified modified_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+ALTER TABLE unknown_claim
+RENAME TO  abnormal_claim,
+  ADD COLUMN created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  ADD COLUMN modified_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+ALTER TABLE claim
+  DROP FOREIGN KEY claim_ibfk_1;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+ALTER TABLE claim
+  CHANGE COLUMN transaction_by_hash_id transaction_hash_id VARCHAR(70) CHARACTER SET 'latin1' COLLATE 'latin1_general_ci' NULL DEFAULT NULL ,
+  CHANGE COLUMN is_n_s_f_w is_nsfw TINYINT(1) NOT NULL DEFAULT '0' ,
+  CHANGE COLUMN created created_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  CHANGE COLUMN modified modified_at DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP;
+-- +migrate StatementEnd
+
+-- +migrate StatementBegin
+ALTER TABLE claim
+  ADD CONSTRAINT claim_ibfk_1
+FOREIGN KEY (transaction_hash_id)
+REFERENCES transaction (hash)
+  ON DELETE CASCADE
+  ON UPDATE NO ACTION;
```

Comment:
why not `ON UPDATE CASCADE`?

Reply:
Maybe, I am misunderstanding the difference, but it was because that should never happen(the update of the hash of a transaction). If it does, it means there is a problem with the data. Instead, no action will produce an error because it will violate the constraint because the child will not have a parent. For example, a claim points to a transaction. I do not want a transaction to be modified such that its hash is modified.  Chainquery doesn't own the uniqueness of a transaction. This is information that comes from LBRYcrd. To keep the data aligned, I wanted these fields set once. Almost all of my foreign key relations are setup this way. It prevents the changing of foreign key columns.